### ISSUE TITLE
Possibility of emitting notifySuccess without closing the auth pop-up window

### ISSUE THREAD

**Issue by samgavinio** [OfficeDev/microsoft-teams-library-js#15]
This is more of a feature request than anything. Would it be possible to add an option to modify the behaviour of the `notifySuccess` and `notifyFailure` functions so that they won't close the auth pop-up when they're called. 

It could be useful for rendering a post authentication message on the pop-up window and then deferring closing the pop-up to the user code.

**Comment by dfederm** [comment]
Would it make sense to just delay the `notifySuccess` or `notifyFailure` calls until the auth popup is ready to close? For example you mention deferring the closing to the user code, but instead of calling window.close, that's the point at which you could call `notifySuccess` or `notifyFailure`.

**Comment by samgavinio** [comment]
Hi, yes but delaying the notify call would mean that the parent window would also have to unnecessarily wait to receive the success event when in fact the authentication flow in the pop-up window has already been completed. Doesn't feel right to have to delay updating the parent window just to render a post authentication message in the popup window.

I'm also thinking of a use-case where an end-user could close the auth window while the notify is being delayed. In this case the parent would never receive the success/failure event.

And, thanks for taking the time to consider this.

**Comment by dfederm** [comment]
Technically this wouldn't be hard to implement, so my concerns are mainly around the design and whether it "makes sense" on the SDK. It adds some complexity, as there would be less of a clear life cycle to the auth popup. We'd have to handle the case where the auth popup replied already but is still open for some reason.

In general around the web the common pattern which the current implementation follows is to open the auth popup, the user logs in, the window self-closes, and the original window updates. I haven't really seen many scenarios where you continue to interact with the auth popup once the auth flow is done.

Regarding the scenario where the user closes the auth popup, we've actually made a change for the next version to detect this and call the failure callback in that case with a specific reason string. So at least we'd be able to detect that. I think scenario-wise we're considering the user closing the popup at any point in the auth flow to be considered a cancellation, mainly because of the common auth flow mentioned above. Basically the user either finishes the entire auth flow, or they don't, and that's tied to the life cycle of the popup. I suppose one could make the argument of a "you are now signed in" message, but IMHO those don't really add any value and it's pretty clear by the popup closing and the page behind it updating that the log in was successful.

Without a concrete example, I'd suggest moving anything that's not part of the auth flow out of the auth popup and into the parent frame.
